diff --git a/include/subdir.mk b/include/subdir.mk
index 6512e24c2e..c54d5cd48c 100644
--- a/include/subdir.mk
+++ b/include/subdir.mk
@@ -38,12 +38,13 @@ subdir_make_opts = \
 # 2: target
 # 3: build type
 # 4: build variant
+# export args array to logs (opde)
 log_make = \
 	 $(if $(call debug,$(1),v),,@)+ \
 	 $(if $(BUILD_LOG), \
 		set -o pipefail; \
 		mkdir -p $(BUILD_LOG_DIR)/$(1)$(if $(4),/$(4));) \
-	$(SCRIPT_DIR)/time.pl "time: $(1)$(if $(4),/$(4))/$(if $(3),$(3)-)$(2)" \
+	$(SCRIPT_DIR)/time.pl "time: $(1)$(if $(4),/$(4))/$(if $(3),$(3)-)$(2) ['$(1)','$(4)','$(3)','$(2)'] " \
 	$$(SUBMAKE) $(subdir_make_opts) $(if $(3),$(3)-)$(2) \
 		$(if $(BUILD_LOG),SILENT= 2>&1 | tee $(BUILD_LOG_DIR)/$(1)$(if $(4),/$(4))/$(if $(3),$(3)-)$(2).txt)
 
diff --git a/scripts/time.pl b/scripts/time.pl
index 6f10170b3b..68f0099373 100755
--- a/scripts/time.pl
+++ b/scripts/time.pl
@@ -54,8 +54,9 @@ else {
 	my ($sec2, $usec2) = gettime();
 	my (undef, undef, $cuser, $csystem) = times();
 
-	printf STDOUT "%s#%.2f#%.2f#%.2f\n",
-		$prefix, $cuser, $csystem,
+	printf STDOUT "%s#%d#%.2f#%.2f#%.2f\n",
+        # export exitcode (opde)
+		$prefix, $exitcode, $cuser, $csystem,
 		($sec2 - $sec) + ($usec2 - $usec) / 1000000;
 
 	$SIG{'INT'} = 'DEFAULT';
