diff --git a/target/sdk/Makefile b/target/sdk/Makefile
index 3b860db93a..1a3752d96f 100644
--- a/target/sdk/Makefile
+++ b/target/sdk/Makefile
@@ -147,11 +147,11 @@ $(BIN_DIR)/$(SDK_NAME).tar.xz: clean
 
 	-rm -f $(SDK_BUILD_DIR)/feeds.conf.default
 	$(if $(BASE_FEED),echo "$(BASE_FEED)" > $(SDK_BUILD_DIR)/feeds.conf.default)
-	if [ -f $(TOPDIR)/feeds.conf ]; then \
-		cat $(TOPDIR)/feeds.conf >> $(SDK_BUILD_DIR)/feeds.conf.default; \
-	else \
-		cat $(TOPDIR)/feeds.conf.default >> $(SDK_BUILD_DIR)/feeds.conf.default; \
-	fi
+	cat $(TOPDIR)/feeds.conf.default >> $(SDK_BUILD_DIR)/feeds.conf.default
+	$(if $(OPDE_PYTHON), \
+		LC_ALL=C.UTF-8 LANG=C.UTF-8 $(OPDE_PYTHON) $(OPDE_BUILDER) -s $(TOPDIR) @hack-sdk $(SDK_BUILD_DIR), \
+	        echo "env 'OPDE_PYTHON' missing... skip to hack sdk" \
+	)
 	$(SED) 's,^# REVISION:=.*,REVISION:=$(REVISION),g' $(SDK_BUILD_DIR)/include/version.mk
 	$(SED) 's,^# SOURCE_DATE_EPOCH:=.*,SOURCE_DATE_EPOCH:=$(SOURCE_DATE_EPOCH),g' $(SDK_BUILD_DIR)/include/version.mk
 	$(SED) '/LINUX_VERMAGIC:=/ { s,unknown,$(LINUX_VERMAGIC),g }' $(SDK_BUILD_DIR)/include/kernel.mk
